osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -256,8 +256,36 @@ public void PostCommand(string text, Channel target = null)
                     JoinChannel(channel);
                     break;
 
+                case "chat":
+                case "msg":
+                case "query":
+                    if (string.IsNullOrWhiteSpace(content))
+                    {
+                        target.AddNewMessages(new ErrorMessage($"Usage: /{command} [user]"));
+                        break;
+                    }
+
+                    // Check if the user has joined the requested channel already.
+                    // This uses the channel name for comparison as the PM user's username is unavailable after a restart.
+                    var privateChannel = JoinedChannels.FirstOrDefault(
+                        c => c.Type == ChannelType.PM && c.Users.Count == 1 && c.Name.Equals(content, StringComparison.OrdinalIgnoreCase));
+
+                    if (privateChannel != null)
+                    {
+                        CurrentChannel.Value = privateChannel;
+                        break;
+                    }
+
+                    var request = new GetUserRequest(content);
+                    request.Success += OpenPrivateChannel;
+                    request.Failure += e => target.AddNewMessages(
+                        new ErrorMessage(e.InnerException?.Message == @"NotFound" ? $"User '{content}' was not found." : $"Could not fetch user '{content}'."));
```

Comment:
This is pretty bad. This code will silently break if the `APIRequest` failure API surface ever changes. Perhaps this is fine (@peppy want your opinion).

Reply:
What do you mean by the API surface? Not too sure how to respond.

To clarify how the failure event is fired: currently it's only sent for cases where there is a failure in the actual request, not a network or (higher level) auth issue. Not sure if this answers or changes your answer.

Reply:
As in right now the exception type is `APIException` which wraps an `InnerException` containing the actual `WebException`. If this is ever changed (and it's a hidden detail to me, so I don't see why it _couldn't_ change), then this code will break.

Reply:
As in right now the exception type is `APIException` which wraps an `InnerException` containing the actual `WebException`. If this is ever changed (and it's a hidden detail to me, so I don't see why it _couldn't_ change), then this code will break.

Reply:
Right. I think it's okay for now since the fail case here is just showing the wrong message (which will probably be pointed out to us at that point).

Definitely something to revisit though - feels weird having an untyped `Exception` sent through the event. If you want to be extra sure, you could include the full `Exception e` spec in the code here, so if/when we change the siganture of `APIFailureHandler` it will trigger a compile time failure.